flask/helpers.py

```diff
@@ -575,12 +585,21 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
             warn('Access %s failed, maybe it does not exist, so ignore etags in '
                  'headers' % filename, stacklevel=2)
 
-        if conditional:
+    if conditional:
+        try:
+            # Added in werkzeug 0.12
+            rv = rv.make_conditional(request, accept_ranges=True,
+                                     complete_length=fsize)
+        except TypeError:
             rv = rv.make_conditional(request)
-            # make sure we don't send x-sendfile for servers that
-            # ignore the 304 status code for x-sendfile.
-            if rv.status_code == 304:
-                rv.headers.pop('x-sendfile', None)
+        except RequestedRangeNotSatisfiable:
+            if close_on_416:
```

Comment:
That's only what I'm inferring from the code. It does close the file in some cases and leave it open in the case where the file object is passed to `make_response`. However, `werkzeug.BaseResponse` appears to have its own mechanism for closing resources.